security-admin/src/main/java/org/apache/ranger/rest/XUserREST.java

```diff
@@ -863,6 +868,49 @@ public void deleteXUserByUserName(@PathParam("userName") String userName,
 		xUserMgr.deleteXUser(vxUser.getId(), forceDelete);
 	}
 
+
+	/**
+	 * Proceed with caution: Force deletes users in bulk from the ranger db,
+	 * essentially serves as a cleanup Op for external users.
+	 * <tt>Delete</tt> happens one at a time with immediate commit on the transaction.
+	 */
+	@DELETE
+	@Path("/delete/external/users")
+	@PreAuthorize("hasRole('ROLE_SYS_ADMIN')")
+	@Produces({ "application/json" })
+	public Response deleteXUsers() {
```

Comment:
Instead of an API specifically to delete all external users, consider a generic API that let's the call specify the users to delete via query-params - similar to in searchXUsers():

```
@DELETE
@Path("/users")
@PreAuthorize("hasRole('ROLE_SYS_ADMIN')")
@Produces({ "application/json" })
public Response deleteUsers(@Context HttpServletRequest request) {
  SearchCriteria searchCriteria = searchUtil.extractCommonCriterias(request, xUserService.sortFields);

  // TODO: update searchCriteria with query-params 

  Long<Long> userIds = xUserMgr.deleteUsers(searchCriteria);

  return Response.ok(userIds.size() + " external users deleted successfully").build();
}

```

Reply:
thanks for the suggestion, changes included in the latest patch.